compiler/test/dotty/tools/debug/DebugStepAssert.scala

```diff
@@ -0,0 +1,138 @@
+package dotty.tools.debug
+
+import com.sun.jdi.Location
+import dotty.tools.io.JPath
+import dotty.tools.readLines
+
+import scala.annotation.tailrec
+
+/**
+ * A debug step and an associated assertion to validate the step.
+ * A sequence of DebugStepAssert is parsed from the check file in tests/debug
+ */
+private[debug] case class DebugStepAssert[T](step: DebugStep[T], assert: T => Unit)(
+  using val location: CheckFileLocation
+)
+
+/** A location in the check file */
+private[debug] case class CheckFileLocation(checkFile: JPath, line: Int):
+  override def toString: String = s"$checkFile:$line"
+
+/** When a DebugStepAssert fails it throws a DebugStepException */
+private[debug] case class DebugStepException(message: String, location: CheckFileLocation) extends Exception
+
+private[debug] enum DebugStep[T]:
+  case Break(className: String, line: Int) extends DebugStep[Location]
+  case Step extends DebugStep[Location]
+  case Next extends DebugStep[Location]
+  case Eval(expression: String) extends DebugStep[Either[String, String]]
+
+private[debug] object DebugStepAssert:
+  import DebugStep.*
+  def parseCheckFile(checkFile: JPath): Seq[DebugStepAssert[?]] =
+    val sym = "[a-zA-Z0-9$.]+"
+    val line = "\\d+"
+    val trailing = s"\\s*(?:\\/\\/.*)?".r // empty or comment
+    val break = s"break ($sym) ($line)$trailing".r
+    val step = s"step ($sym|$line)$trailing".r
+    val next = s"next ($sym|$line)$trailing".r
+    val multiLineEval = s"eval$trailing".r
+    val eval = s"eval (.*)".r
+    val result = "result (.*)".r
+    val error = "error (.*)".r
+    val multiLineError = s"error$trailing".r
```

Comment:
All those regexes should be defined in the `object`, not in the `def`. Otherwise they need to be recompiled every time.

Reply:
done